### ISSUE TITLE
Can't encrypt Pathnames

### ISSUE THREAD

**Issue by madblobfish** [metaware/underlock#3]
When encrypting a pathname it fails or loads the file contents. The later one can be unwanted.

Example:
```ruby
require 'underlock'
require 'pathname' # you should require this in your library
key = OpenSSL::PKey::RSA.new 200
Underlock::Base.configure do |config|
  config.public_key  = key.to_pem
  config.private_key = key.public_key.to_pem
  config.cipher      = OpenSSL::Cipher.new('aes-256-gcm')
end
Underlock::Base.encrypt('.')
# fails with:  Errno::EISDIR: Is a directory @ io_fread
```

**Comment by jasdeepsingh** [comment]
@madblobfish thanks for reporting this issue, It was never intended for this library to encrypt complete directories, when passed a string to encrypt, we check if it could be a valid path and if it is we try to encrypt the file at the location. Looks like we are not handling the scenario if the path points to a directory. 

I'll push a fix for this today.

**Comment by madblobfish** [comment]
What if a user wants to encrypt a string representing a path, but the library then tries to load the contents...
I'd propose turning this into an opt-in feature as it is not obvious.

**Comment by madblobfish** [comment]
When encrypting a string which is the Pathname of a existing file it will load the contents and not encrypt the sting this behavior can be unwanted.
I'd propose to make the Pathname check opt-in with a option or something.

**Comment by jasdeepsingh** [comment]
@madblobfish makes sense, and I agree with the proposition. I think I'd like to make the argument type be the judge of what happens. If a user wants to encrypt a file, they explicitly pass a `Pathname` or `File` object, if it's just a `String`, we encrypt just the string.

**Comment by jasdeepsingh** [comment]
@madblobfish closing this issue, because this is fixed in `v0.0.4`: https://github.com/metaware/underlock/releases/tag/v0.0.4
